CMakeLists.txt

```diff
@@ -220,152 +170,63 @@ endif()
-  endif()
-
-
-endif(${BUILD_TESTS})
-
-# copy reference data that might be used in the example
-add_custom_target(copy_reference_data ALL
-    DEPENDS
-    copy_test_reference_data
-  )
-
-add_custom_command(OUTPUT copy_test_reference_data
-    COMMAND ${CMAKE_COMMAND} -E copy_directory
-    ${CMAKE_SOURCE_DIR}/tests/reference_data ${CMAKE_BINARY_DIR}/reference_data
-    DEPENDS
-        always_rebuild
-  )
-
-# put it there to avoid warnings from boost tests
-add_compile_options(-Werror) # warnings are errors!
+enable_testing()
+if(BUILD_TESTS)
+    add_subdirectory(tests)
+endif()
 
-# compile the library
-add_subdirectory(
-  ${CMAKE_SOURCE_DIR}/src/
-  )
+if(BUILD_EXAMPLES)
+    add_subdirectory(examples)
+endif()
 
-# set up bindings
 if(BUILD_BINDINGS)
-add_subdirectory(
-  ${CMAKE_SOURCE_DIR}/bindings/
-)
+    add_subdirectory(bindings)
 endif()
 
-# set up documentation
-if (${ENABLE_DOC})
-  add_subdirectory(
-    ${CMAKE_SOURCE_DIR}/docs/
-    )
-endif (${ENABLE_DOC})
-
-if (${BUILD_EXAMPLES})
-  # compile exmaples executables
-  file( GLOB binaries "${CMAKE_SOURCE_DIR}/examples/*.cc")
-  foreach(binaryfile ${binaries})
-    get_filename_component(binaryname  ${binaryfile} NAME_WE)
-    add_executable(${binaryname} ${binaryfile})
-    target_link_libraries(${binaryname} "${LIBRASCAL_NAME}")
-  endforeach(binaryfile ${binaries})
-  configure_file("${CMAKE_SOURCE_DIR}/examples/reference_data/crystal_structure.json" ${CMAKE_BINARY_DIR} COPYONLY)
-  configure_file("${CMAKE_SOURCE_DIR}/examples/reference_data/alanine-X.json" ${CMAKE_BINARY_DIR} COPYONLY)
-  configure_file("${CMAKE_SOURCE_DIR}/tests/reference_data/alanine-center-select.json" ${CMAKE_BINARY_DIR} COPYONLY)
-
-  # Add profiling information for the appropriate examples
-  if (ENABLE_PROFILING)
-    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-    target_compile_options("soap_profile" PRIVATE -ggdb)
-      # target_compile_options("soap_profile" PRIVATE -pg)
-      # set_target_properties("soap_profile" PROPERTIES LINK_FLAGS -pg)
-    endif()
-  endif()
-endif (${BUILD_EXAMPLES})
-
-
-# or copy them
-file (GLOB pybins "${CMAKE_SOURCE_DIR}/bin/*.py")
-foreach(pybin ${pybins})
-  get_filename_component(binaryname  ${pybin} NAME_WE)
-  configure_file(
-    ${pybin}
-    "${CMAKE_BINARY_DIR}/${binaryname}.py"
-    COPYONLY)
-endforeach(pybin ${pybins})
+if(BUILD_DOC)
+    add_subdirectory(docs)
+endif()
 
+# Try to find optional cpp linter
+include(cpplint)
 # Checking hh/cc files for compliance with coding conventions
 if(CPPLINT_FOUND)
   # register the folders to check for cpp code conventions compliance
   cpplint_add_subdirectory("${CMAKE_SOURCE_DIR}/src" "")
   cpplint_add_subdirectory("${CMAKE_SOURCE_DIR}/tests" "")
   cpplint_add_subdirectory("${CMAKE_SOURCE_DIR}/bindings" "")
-  cpplint_add_subdirectory("${CMAKE_SOURCE_DIR}/examples"
-  "--filter=-build/namespaces")
+  cpplint_add_subdirectory("${CMAKE_SOURCE_DIR}/examples" "--filter=-build/namespaces")
```

Comment:
Any reason for making this one line instead of the break?

Reply:
I like it better :smiley:!  :man_shrugging:

But this flag is actually not needed: the code in examples already uses //NOLINT for using namespace XXX